dulwich/porcelain.py

```diff
@@ -170,52 +174,53 @@ def open_repo_closing(path_or_repo):
     return closing(Repo(path_or_repo))
 
 
-def path_to_tree_path(repopath, path):
+def path_to_tree_path(repo_path: str, tree_path: str) -> bytes:
     """Convert a path to a path usable in e.g. an index.
 
-    :param repo: Repository
-    :param path: A path
-    :return: A path formatted for use in e.g. an index
+    :param repo_path: Absolute path to the repository
+    :param tree_path: Path to a file in the repository, not necessarily absolute
+    :return: A relative path formatted for use in e.g. an index
     """
-    os.path.relpath(path, repopath)
+    if tree_path.startswith(repo_path):
+        tree_path = os.path.relpath(tree_path, repo_path)
     if os.path.sep != '/':
-        path = path.replace(os.path.sep, '/')
-    return path.encode(sys.getfilesystemencoding())
+        tree_path = tree_path.replace(os.path.sep, '/')
+    return tree_path.encode(sys.getfilesystemencoding())
 
 
-def archive(repo, committish=None, outstream=default_bytes_out_stream,
-            errstream=default_bytes_err_stream):
+def archive(repo: Union[BaseRepo, str], commit_id: Union[str, bytes]=None,
+            out_stream=default_bytes_out_stream, err_stream=default_bytes_err_stream):
     """Create an archive.
 
-    :param repo: Path of repository for which to generate an archive.
-    :param committish: Commit SHA1 or ref to use
-    :param outstream: Output stream (defaults to stdout)
-    :param errstream: Error stream (defaults to stderr)
+    :param repo: Path of repository or repo object for which to generate an archive.
+    :param commit_id: Commit SHA1 or ref to use
```

Comment:
The name committish ish intentional; it's the term Git uses itself, and it can be either a commit id *or* a ref name.